### PR TITLE
nix-selinux-policy: init at 1.0

### LINKED ISSUE #2374: Nix installer fails on Fedora 28 due to SELinux denial
I have just come across this trying to install Nix on Fedora 28 Workstation with SELinux enabled (default) using the `./install-multi-user` script from the `nix-2.0.4-x86_64-linux` bundle.

```
---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo systemctl link /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.service                                                                                

to set up the nix-daemon service

Failed to link unit: Access denied
```

I can confirm this is an SELinux denial due to this AVC message in `/var/log/audit/audit.log`:
```
type=AVC msg=audit(1535267537.912:2983): avc:  denied  { read } for  pid=1 comm="systemd" name="default" dev="nvme0n1p7" ino=2095622 scontext=system_u:system_r:init_t:s0 tcontext=unconfined_u:object_r:default_t:s0 tclass=lnk_file permissive=0 
```

This issue seems to be the same as the one [discussed](https://logs.nix.samueldr.com/nixos-dev/2018-04-12) in #nixos-dev.

Based on the nature of the error, I would naturally assume all SELinux-enabled systems would be affected by this.

### PR DESCRIPTION
**Pull request by outergod**
###### Motivation for this change

This basic SELinux policy for files in `/nix` is required to support running Nix in multi-user mode on SELinux-enabled systems.

To test the output of this derivation, run `semodule -i result/nix.pp` after building and `restorecon -v` on files in `/nix` to verify that the contexts defined in `nix.fc` are applied.

Required to fix https://github.com/NixOS/nix/issues/2374.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jtojnar** [comment]
Why not add it to Nix itself?

**Comment by e-user** [comment]
@jtojnar I was considering just that. There are pros and cons: On the one hand, this SELinux module is directly tied to nix-store and what it does. On the other hand, decoupling it from the Nix project itself means it's much easier to develop and release separately. A simple tweak on the policy would mean that all of Nix would have to be re-compiled, the version bumped and released, even though nothing changed in Nix itself. The same holds true vice versa: Version bumps in Nix don't mean that the policy needs to be changed, rebuild and deployed.

Lastly, this is also what Fedora does: Most policies are pre-integrated with the OS, and special needs policies for specific packages come as separately versioned and built packages.

File: pkgs/os-specific/linux/nix-selinux-policy/LICENSE [hunk 1]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (edolstra) [inline pkgs/os-specific/linux/nix-selinux-policy/LICENSE]: IMHO individual Nix expressions in Nixpkgs should not have different licenses.

**Comment by jtojnar** [comment]
IMHO the main criterion to decide whether to upstream the file is whether it can be reused by other distributions like desktop files, manual pages or appstream metadata can.

File: pkgs/os-specific/linux/nix-selinux-policy/LICENSE (hunk 1 above)
Comment (e-user) [reply pkgs/os-specific/linux/nix-selinux-policy/LICENSE]: Custom license removed, and since there's no upstream project, I applied nixpkgs' MIT license.

**Comment by e-user** [comment]
@jtojnar the policy, in fact, is currently only of use for other distributions than NixOS. However it has to be loaded into the host system's SELinux configuration and there is no mechanism to do this yet except for the one I added to Nix' installer in NixOS/nix#2670, plus files have to be re-labeled unless they had been previously been installed by Nix.

Keeping the policy separate from Nix has purely pragmatic advantages: It can thus be modified without requiring a separate release of Nix. 

The effect of the policy, once fully integrated, is global: It affects the future labeling of all files in the Nix store and makes it at least possible to run systemd services based on files from the Nix store on SELinux systems. 

I should've probably explained this much earlier, in the description. In any case, the resulting file has to be inserted using `semodule -i`, first. The mechanism to apply newer versions of the policy should probably also be added to Nix later, as labels have to be restored retroactively.

If you still think it shouldn't go to nixpkgs but Nix, I'm perfectly fine with closing the PR and moving the policy to the aforementioned PR to Nix. However hosting the policy with Nix will complicate future updates to the policy, and I think there are no advantages in not having the policy in nixpkgs except for purely idealistic ones.

**Comment by e-user** [comment]
I ultimately decided to move the policy into the related Nix PR itself.

**Pull request closed without merging**